common/app/model/dotcomrendering/pageElements/PageElement.scala

```diff
@@ -867,27 +867,40 @@ object PageElement {
         val imageAssets = element.assets.zipWithIndex
           .map { case (a, i) => ImageAsset.make(a, i) }
 
-        val imageSources: Seq[ImageSource] = DotcomRenderingImageRoleWidthByBreakpointMapping.all.map {
-          case (weighting, widths) =>
-            val srcSet: Seq[SrcSet] = widths.breakpoints.flatMap { b =>
-              Seq(
-                ImgSrc.srcsetForBreakpoint(
-                  b,
-                  DotcomRenderingImageRoleWidthByBreakpointMapping.immersive.breakpoints,
-                  maybeImageMedia = Some(ImageMedia(imageAssets)),
-                ),
-                ImgSrc.srcsetForBreakpoint(
-                  b,
-                  DotcomRenderingImageRoleWidthByBreakpointMapping.immersive.breakpoints,
-                  maybeImageMedia = Some(ImageMedia(imageAssets)),
-                  hidpi = true,
-                ),
-              )
-            }.flatten
-            // A few very old articles use non-https hosts, which won't render
-            val httpsSrcSet = srcSet.map(set => set.copy(src = ensureHTTPS(set.src)))
-            ImageSource(weighting, httpsSrcSet)
-        }.toSeq
+        def getImageBreakpointMapping(
+            isMainMedia: Boolean,
+            isImmersive: Boolean,
+        ): ImageRoleWidthsByBreakpointMapping = {
+          if (isMainMedia) MainMedia
+          else if (isImmersive) ImmersiveMedia
+          else BodyMedia
+        }
+
+        def makeImageSources(breakPointMapping: ImageRoleWidthsByBreakpointMapping): Seq[ImageSource] = {
+          breakPointMapping.all.map {
+            case (weighting, widths) =>
+              val srcSet: Seq[SrcSet] = widths.breakpoints.flatMap { b =>
+                Seq(
+                  ImgSrc.srcsetForBreakpoint(
+                    b,
+                    breakPointMapping.immersive.breakpoints,
+                    maybeImageMedia = Some(ImageMedia(imageAssets)),
+                  ),
+                  ImgSrc.srcsetForBreakpoint(
+                    b,
+                    breakPointMapping.immersive.breakpoints,
+                    maybeImageMedia = Some(ImageMedia(imageAssets)),
+                    hidpi = true,
+                  ),
```

Comment:
I'm still not sure why it always uses `.immersive.breakpoints`, it was like this in the existing code too - would love some suggestions as to why this is!

Reply:
I looked into this once (when I had [a branch](https://github.com/guardian/frontend/compare/js-image-sources) with the same change you're making 😆 )

It's a bit convoluted but I don't think it's actually used. The code path is something like:
- `ImgSrc.srcsetForBreakpoint` [uses that argument](https://github.com/guardian/frontend/blob/5b970cd7308175cfc1bcae2d4fb8c06ee13c5fa0/common/app/views/support/ImageProfile.scala#L393) to pass to `toPixels`
- `toPixels` only uses those `breakpointWidths` in one `case` of the [pattern match](https://github.com/guardian/frontend/blob/5b970cd7308175cfc1bcae2d4fb8c06ee13c5fa0/common/app/layout/BreakpointWidth.scala#L15); where the widths in the given `BreakpointWidth` are not already pixel values (and we're on the mobile breakpoint
- None of the `ContentWidth` [breakpoints we are using here](https://github.com/guardian/frontend/blob/5b970cd7308175cfc1bcae2d4fb8c06ee13c5fa0/common/app/layout/ContentWidths.scala#L39) has anything but pixel based widths (AFAICT)

I think this code could really do with simplifying, it's quite confusing and I'm not sure it's all even used (e.g. `toPixels` seems largely redundant unless we have non-pixel-based widths somewhere)

Reply:
Having a quick look, `ImgSrc.srcsetForBreakpoint` is used about 8 times across the app, including in the commercial app ~ I'm not confident we could remove the `breakpointWidths` parameter as it's not as clear that all breakpoints will be in pixels. I found that the [FaciaWidths](https://github.com/guardian/frontend/blob/main/common/app/layout/FaciaWidths.scala#L10) class has one use of `vw` instead of `px`.